From a2d90ecabf3bcf3aa22ed9dbbecfd3d37902956f Mon Sep 17 00:00:00 2001
From: Damien Regad <dregad@mantisbt.org>
Date: Tue, 7 Mar 2017 12:34:16 +0100
Subject: [PATCH] Fix XSS in bug_change_status_page.php

The value of the change_type parameter was not encoded before being
displayed as a hidden input.

This vulnerability was reported by Etienne Landais.

Fixes #22486
---
 bug_change_status_page.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bug_change_status_page.php b/bug_change_status_page.php
index 7e465cf846..12cf01a279 100644
--- a/bug_change_status_page.php
+++ b/bug_change_status_page.php
@@ -154,7 +154,7 @@
 	<input type="hidden" name="bug_id" value="<?php echo $f_bug_id ?>" />
 	<input type="hidden" name="status" value="<?php echo $f_new_status ?>" />
 	<input type="hidden" name="last_updated" value="<?php echo $t_bug->last_updated ?>" />
-	<input type="hidden" name="action_type" value="<?php echo $f_change_type; ?>" />
+	<input type="hidden" name="action_type" value="<?php echo string_attribute( $f_change_type ); ?>" />
 
 <?php
 	$t_current_resolution = $t_bug->resolution;
